wallet/wallet.c

```diff
@@ -2158,3 +2158,46 @@ struct outpoint *wallet_outpoint_for_scid(struct wallet *w, tal_t *ctx,
 
 	return op;
 }
+
+void wallet_transaction_add(struct wallet *w, const struct bitcoin_tx *tx,
+			    const u32 blockheight, const u32 txindex)
+{
+	struct bitcoin_txid txid;
+	sqlite3_stmt *stmt = db_prepare(w->db, "SELECT blockheight FROM transactions WHERE id=?");
+	bool known;
+
+	bitcoin_txid(tx, &txid);
+	sqlite3_bind_sha256(stmt, 1, &txid.shad.sha);
+	known = sqlite3_step(stmt) == SQLITE_ROW;
+	sqlite3_finalize(stmt);
```

Comment:
This seems like a debugging check, in which case it'd be better to check that the other fields haven't changed?  Plus add a /*FIXME: change to INSERT OR UPDATE */ for later optimization?

Reply:
I'm not sure I get the question. I could change this to become `SELECT count(*) FROM transactions WHERE id=?` and then check that I get a 1 as a result, but the effect is the same, either I have a row or I don't, and depending on which it is I need to insert or update.

Or are you saying we should fetch all fields and check that they are identical?